Sight_++_Solution/Intel_Sight_PP/src/caffe_impl.cpp

```diff
@@ -50,68 +54,78 @@ struct CaffeModelImpl : public ModelInterface {
-				auto y_left_bottom = static_cast<int>(detection_matrix.at<float>(i, 4) * color_matrix.rows);
-
-				auto x_right_top = static_cast<int>(detection_matrix.at<float>(i, 5) * color_matrix.cols);
-				auto y_right_top = static_cast<int>(detection_matrix.at<float>(i, 6) * color_matrix.rows);
-
-				cv::Rect object(
-					static_cast<int>(x_left_bottom), static_cast<int>(y_left_bottom),
-					static_cast<int>(x_right_top - x_left_bottom), static_cast<int>(y_right_top - y_left_bottom));
-				object = object & cv::Rect(0, 0, depth_matrix.cols, depth_matrix.rows);
-				// Calculate valid depth inside the detection region
-				cv::Mat object_depth = depth_matrix(object);
-
-				cv::Mat pixel_array;
-
-				object_depth.convertTo(pixel_array, CV_32F);
-				pixel_array = pixel_array.reshape(1, pixel_array.total());
-				cv::Mat label, centers;
-
-				kmeans(pixel_array, 2, label,
-					cv::TermCriteria(cv::TermCriteria::COUNT + cv::TermCriteria::EPS, 10, 0.1), 1,
-					cv::KMEANS_PP_CENTERS, centers);
-
-
-				// replace pixel values with their center value:
-				float* p = pixel_array.ptr<float>();
-				int group_zero = 0;
-				int group_one = 0;
-				for (size_t i = 0; i < pixel_array.rows; i++) {
-					int center_id = label.at<int>(i);
-					if (center_id == 0) {
-						group_zero++;
-
+				auto confidence = detection_matrix.at<float>(id, 2);
+
+				if (confidence > confidence_threshold)
+				{
+					auto object_class = static_cast<size_t>(detection_matrix.at<float>(id, 1));
+
+					auto x_left_bottom = static_cast<int>(detection_matrix.at<float>(id, 3) * color_matrix.cols);
+					auto y_left_bottom = static_cast<int>(detection_matrix.at<float>(id, 4) * color_matrix.rows);
+
+					auto x_right_top = static_cast<int>(detection_matrix.at<float>(id, 5) * color_matrix.cols);
+					auto y_right_top = static_cast<int>(detection_matrix.at<float>(id, 6) * color_matrix.rows);
+
+					cv::Rect object(
+						static_cast<int>(x_left_bottom), static_cast<int>(y_left_bottom),
+						static_cast<int>(x_right_top - x_left_bottom), static_cast<int>(y_right_top - y_left_bottom));
+					object = object & cv::Rect(0, 0, depth_matrix.cols, depth_matrix.rows);
+					// Calculate valid depth inside the detection region
+					cv::Mat object_depth = depth_matrix(object);
+
+					cv::Mat pixel_array;
+
+					object_depth.convertTo(pixel_array, CV_32F);
+					pixel_array = pixel_array.reshape(1, pixel_array.total());
+					cv::Mat label, centers;
+
+					kmeans(pixel_array, 2, label,
+						cv::TermCriteria(cv::TermCriteria::COUNT + cv::TermCriteria::EPS, 10, 0.1), 1,
+						cv::KMEANS_PP_CENTERS, centers);
+
+
+					// replace pixel values with their center value:
+					float* p = pixel_array.ptr<float>();
+					int group_zero = 0;
+					int group_one = 0;
+					for (size_t i = 0; i < pixel_array.rows; i++) {
+						int center_id = label.at<int>(i);
+						if (center_id == 0) {
+							group_zero++;
```

Comment:
7 indents is a lot 😃

Reply:
How can I improve it

Reply:
One is to flip the confidence if statement.
```c++
if(confidence > confidence_threshold) {
  work on confident objects
```
```c++
if(confidence < confidence_threshold) return;
work on confident objects
```
Then you always put things into functions.
But, its okay for now 😄